web/src/components/Dialogs/ImageDialog.tsx

```diff
@@ -0,0 +1,121 @@
+import { AnimatePresence, motion } from "framer-motion";
+import Image from "next/future/image";
+import { useEffect, useState } from "react";
+import { MdFavorite, MdFavoriteBorder, MdLibraryAdd } from "react-icons/md";
+
+import Button from "../Inputs/Button";
+import IconButton from "../Inputs/IconButton";
+import Card from "../Surfaces/Card";
+import Author from "../Widgets/Author";
+import LikesCounter from "../Widgets/LikesCounter";
+import StyledDialog from "./StyledDialog";
+
+interface Props {
+  src: string;
+  prompt: string;
+  likes: number;
+  isLiked?: boolean;
+  authorName: string;
+  authorAvatar: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export default function ImageDialog({
+  src,
+  prompt,
+  likes,
+  isLiked = false,
+  authorName,
+  authorAvatar,
+  isOpen,
+  onClose,
+}: Props) {
+  //#region Hooks
+
+  const [isLikedLocally, setIsLikedLocally] = useState(isLiked);
+  const [likesLocally, setLikesLocally] = useState(likes);
+
+  useEffect(() => {
+    setIsLikedLocally(isLiked);
+  }, [isLiked]);
+
+  useEffect(() => {
+    setLikesLocally(likes);
+  }, [likes]);
```

Comment:
This feels a little strange. I think it would be better to just lift all of the state outside of the component and provide an `onLikedChange`. That way this can just be a dumb data component and any data fetching or state management can be handled by someone who actually knows what to do.